### DISCUSSION TITLE
Ossec Agent - tmp file corrupt

### DISCUSSION THREAD

**Discussion by HBA-EMK** [Security-Onion-Solutions/securityonion#8360] [Unsupported Versions]
Good day,

I have a standalone implementation for which I'm installing all 3 agents on each endpoint.

I have replaced the self signed certificate for the manager with a local CA signed one.

After the change, winlogbeat worked flawlessly. I had to tweak the OsQuery agent for it to work (replacing the roots.pem file with the CA certificate). But couldn't get Wazuh to work.

I don't know if it's linked to the certificate change, but now the logs loop on the following message : 

ossec-agent: ERROR: Temporary path file 'tmp/tmp_165848618521904' is corrupt: missing line end. 

The tmp folder is empty. The logs change if I delete it. Gave full control to "everyone" just in case an access issue. The problem has been flagged on all agents and I could hardly find any documentation on the error.

Can you please help me with that?

**Comment by dougburks** [comment]
> all 3 agents on each endpoint.

Is there a particular reason for installing all 3 agents on each endpoint? This will likely result in feature overlap and log duplication and could also result in performance or other issues. We recommend choosing 1 of the agents:
https://docs.securityonion.net/en/2.3/host.html

> ossec-agent: ERROR: Temporary path file 'tmp/tmp_165848618521904' is corrupt: missing line end.

Take a look at https://github.com/wazuh/wazuh/issues/5689 and see if it gives you any additional clues.

You might be able to stop the agent and delete the tmp file.

**Reply by HBA-EMK** [reply]
To be fair, I never challenged the duplicate logs aspect which might be relevant to my context.
I use Wazuh for HIDS, OsQuery to take advantage of the low level API queries (just noticed it does log collection as well), and Beats for log collection and eventual correlation for file tracing.

I did check the issue 5689, In my case however the tmp files are not even created when the log message appears. The folder is most of the time empty (confirmed right now, the error message is looping on different file names, folder still empty).
I changed the Wazuh configuration to communicate with a different standalone SO manager (test environment), I get the same issue, but since this environment only uses the built-in certificates I think the latter are unrelated.
Purging and re-installing the agent doesn't fix the issue. 

I also noticed the following message in the logs : 

2022/07/25 12:46:38 ossec-agent: ERROR: Could not get message for (compliance)

**Comment by btdewey** [comment]
I agree that having all three agents on a system is a lot of duplication.  It's possible to overload endpoints with agents and then they slow down a great deal.  Wazuh and OsQuery provide additional capabilities other than just log shipping so I would probably drop Beats as a complete duplicate.  You might be able to configure only one to send OS logs to reduce duplicate logging, but I'm not familiar enough with them to know how to do that.  That doesn't help with the error unfortunately.  What OS version are you installing this on?
